### PR TITLE
Filter by tag 2

### PR DESCRIPTION
**Pull request by charliew-rsi**
Additional functionality added to get filters rigged correctly through Redux store and filter articles using said tags without modifying the articles list in the store.

### CODE REVIEW

File: public/styles/Register.css [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
   color: #ff0000;
   font-size: 0.75rem;
 }
 
+.float-right {
+  float: right;
+}
+
+br {
+  line-height: 1.5;
```
Comment (wesdollar) [inline public/styles/Register.css]: 1.5 what?

File: src/components/Home/searchBar.jsx [hunk 2]
```diff
@@ -1,43 +1,99 @@
...
-          type="text"
-          name="searchInput"
-          onChange={event => {
-            this.inputField(event);
-          }}
-          placeholder="Enter Search parameters"
-        />
-        <input type="button" onClick={this.search} value="search" />
+        <div className="row">
+          {error.length > 0 ? <div className="error">{error}</div> : <br />}
```
Comment (wesdollar) [inline src/components/Home/searchBar.jsx]: No need for the `> 0`.

File: src/components/ArticleList.js [hunk 3]
```diff
@@ -1,37 +1,29 @@
...
+    return <div className="article-preview">Loading...</div>;
   }
 
-  if (props.articles.length === 0) {
-    return (
-      <div className="article-preview">
-        No articles are here... yet.
-      </div>
-    );
+  if (articles.length === 0) {
```
Comment (wesdollar) [inline src/components/ArticleList.js]: Should be written as `!articles.length`

File: src/reducers/tags.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      return {
+        ...state,
+        tags: action.payload
+      };
+
+    case CHECK_TAG:
+      let tagArr = state.tags;
+      let { articles } = state.articleList;
+      tagArr[action.payload.index].selected = action.payload.value;
+      //let articles = state.articleList.articles.filter((article) => {
```
Comment (wesdollar) [inline src/reducers/tags.js]: Remove commented out code or leave comment why code needs to remain

File: public/styles/Register.css (hunk 1 above)
Comment (charliew-rsi) [reply public/styles/Register.css]: That is a good question. The style was borrowed from the original css for consistency. Further testing shows that browser evaluates the implicit term as rem, thus displaying as intended. Modifying the css to be explicit.

File: src/components/Home/searchBar.jsx (hunk 2 above)
Comment (charliew-rsi) [reply src/components/Home/searchBar.jsx]: Revising

File: src/components/ArticleList.js (hunk 3 above)
Comment (charliew-rsi) [reply src/components/ArticleList.js]: revising

File: src/reducers/tags.js (hunk 4 above)
Comment (charliew-rsi) [reply src/reducers/tags.js]: removing

**Pull request merged**